src/lib/prover/prover.ml

```diff
@@ -83,144 +83,146 @@ module Worker_state = struct
-                 don't_wait_for
-                 @@ Internal_tracing.toggle ~logger
-                      (if enabled then `Enabled else `Disabled)
-
-               let set_itn_logger_data ~daemon_port =
-                 Itn_logger.set_data ~process_kind:"prover" ~daemon_port
-             end : S )
-         | Check ->
-             ( module struct
-               module Transaction_snark = Transaction_snark
-
-               let extend_blockchain (chain : Blockchain.t)
-                   (next_state : Protocol_state.Value.t)
-                   (block : Snark_transition.value) (t : Ledger_proof.t option)
-                   state_for_handler pending_coinbase =
-                 let t, _proof = ledger_proof_opt next_state t in
-                 let res =
-                   Blockchain_snark.Blockchain_snark_state.check ~proof_level
-                     ~constraint_constants
-                     { transition = block
-                     ; prev_state = Blockchain_snark.Blockchain.state chain
-                     ; prev_state_proof =
-                         Lazy.force Mina_base.Proof.blockchain_dummy
-                     ; txn_snark = t
-                     ; txn_snark_proof =
-                         Lazy.force Mina_base.Proof.transaction_dummy
-                     }
-                     ~handler:
-                       (Consensus.Data.Prover_state.handler state_for_handler
-                          ~constraint_constants ~pending_coinbase )
-                     next_state
-                   |> Or_error.map ~f:(fun () ->
-                          Blockchain_snark.Blockchain.create ~state:next_state
-                            ~proof:(Lazy.force Mina_base.Proof.blockchain_dummy) )
-                 in
-                 Or_error.iter_error res ~f:(fun e ->
-                     [%log error]
-                       ~metadata:[ ("error", Error_json.error_to_yojson e) ]
-                       "Prover threw an error while extending block: $error" ) ;
-                 Async.Deferred.return res
-
-               let verify _state _proof = Deferred.return (Ok ())
-
-               let toggle_internal_tracing _ = ()
-
-               let set_itn_logger_data ~daemon_port:_ = ()
-             end : S )
-         | None ->
-             ( module struct
-               module Transaction_snark = Transaction_snark
-
-               let extend_blockchain _chain next_state _block _ledger_proof
-                   _state_for_handler _pending_coinbase =
-                 Deferred.return
-                 @@ Ok
-                      (Blockchain_snark.Blockchain.create
-                         ~proof:(Lazy.force Mina_base.Proof.blockchain_dummy)
-                         ~state:next_state )
-
-               let verify _ _ = Deferred.return (Ok ())
-
-               let toggle_internal_tracing _ = ()
-
-               let set_itn_logger_data ~daemon_port:_ = ()
-             end : S )
-       in
-       m )
+    let m =
```

Comment:
I know it was there already but since the wrapping `Deferred.return` is gone, could we get rid of the `let m = ... in m` "anti-pattern"

Reply:
Ok